lwgenerate/aws/aws.go

```diff
@@ -0,0 +1,383 @@
+func WithLaceworkProfile(name string) AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.LaceworkProfile = name
+	}
+}
+
+// Set the bucket ARN of an existing Cloudtrail setup
+func ExistingCloudtrailBucketArn(arn string) AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.ExistingCloudtrailBucketArn = arn
+	}
+}
+
+// Set the SNS Topic ARN of an existing Cloudtrail setup
+func ExistingSnsTopicArn(arn string) AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.ExistingSnsTopicArn = arn
+	}
+}
+
+// Enable Consolidated Cloudtrail use
+func UseConsolidatedCloudtrail() AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.ConsolidatedCloudtrail = true
+	}
+}
+
+// Set the S3 ForceDestroy parameter to true for newly created buckets
+func EnableForceDestroyS3Bucket() AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.ForceDestroyS3Bucket = true
+	}
+}
+
+// Set an existing IAM role configuration to use with the created Terraform code
+func UseExistingIamRole(iamDetails *ExistingIamRoleDetails) AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.ExistingIamRole = iamDetails
+	}
+}
+
+// Supply additional AWS Profiles to integrate
+func WithSubaccounts(subaccounts ...AwsSubAccount) AwsTerraformModifier {
+	return func(c *GenerateAwsTfConfigurationArgs) {
+		c.SubAccounts = subaccounts
+	}
+}
+
+// Generate new Terraform code based on the supplied args.
+func (args *GenerateAwsTfConfigurationArgs) Generate() (string, error) {
+	// Validate inputs
+	if err := args.validate(); err != nil {
+		return "", errors.Wrap(err, "invalid inputs")
+	}
+
+	// Create blocks
+	requiredProviders, err := createRequiredProviders()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate required providers")
+	}
+
+	awsProvider, err := createAwsProvider(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws provider")
+	}
+
+	laceworkProvider, err := createLaceworkProvider(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate lacework provider")
+	}
+
+	configModule, err := createConfig(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws config module")
+	}
+
+	cloudTrailModule, err := createCloudtrail(args)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate aws cloudtrail module")
+	}
+
+	// Render
+	hclBlocks := lwgenerate.CreateHclStringOutput(
+		lwgenerate.CombineHclBlocks(
+			requiredProviders,
+			awsProvider,
+			laceworkProvider,
+			configModule,
+			cloudTrailModule),
+	)
+	return hclBlocks, nil
+}
+
+func createRequiredProviders() (*hclwrite.Block, error) {
+	return lwgenerate.CreateRequiredProviders(
+		lwgenerate.NewRequiredProvider("lacework",
+			lwgenerate.HclRequiredProviderWithSource("lacework/lacework"),
+			lwgenerate.HclRequiredProviderWithVersion("~> 0.3")))
```

Comment:
We are already in [version `0.12.X` ](https://registry.terraform.io/providers/lacework/lacework/latest)- let us bump it already :)
```suggestion
			lwgenerate.HclRequiredProviderWithVersion("~> 0.12")))
```

Reply:
👍